### PR TITLE
Remove time crate from almost everything

### LINKED ISSUE #7353: Upgrade to time 0.2
The `time` crate has substantially changed between 0.1 and 0.2. This [issue](https://github.com/time-rs/time/issues/197) describes some of the changes. The change that most impacts us is the removal of [`TimeSpec`](https://docs.rs/time/0.1.42/time/struct.Timespec.html). This is a fairly simple struct so we should be able to replicate its functionality. 

See this [issue](https://github.com/time-rs/time/issues/201) for help in converting format strings.

### PR DESCRIPTION
**Pull request by christophermaier**
Rather than updating everything to use the new 0.2 line of the `time` crate, it turns out that we can basically remove it from everything with a bit of refactoring. Juggling multiple time representations was a bit of a pain internally, and made things more confusing than necessary.

This PR performs the (mostly) mechanical translation from `time` to `std::time`, with interesting diversions pointed out along the way. A bit of internal refactoring and cleanup helped with this as well.

Fixes #7353

Signed-off-by: Christopher Maier <cmaier@chef.io>

### CODE REVIEW

**Comment by davidMcneil** [comment]
It looks like there still might be dependencies on [`time 0.1.42`](https://github.com/habitat-sh/habitat/blob/09333bcf5361c217e33dcc62fbc5d954870b83e0/Cargo.lock#L1536). Did only the `launcher` update?

**Comment by christophermaier** [comment]
@davidMcneil :thinking: It might be due to transitive dependencies on the `time` crate... I'll look into it.

File: components/core/src/crypto/keys.rs [hunk 1]
```diff
@@ -309,15 +309,8 @@ fn mk_key_filename<P, S1, S2>(path: P, keyname: S1, suffix: S2) -> PathBuf
...
-    let now = time::now_utc();
-    // https://github.com/rust-lang-deprecated/time/blob/master/src/display.rs
-    // http://man7.org/linux/man-pages/man3/strftime.3.html
-    match now.strftime("%Y%m%d%H%M%S") {
-        Ok(result) => Ok(result.to_string()),
-        Err(_) => Err(Error::CryptoError("Can't parse system time".to_string())),
-    }
-}
+// TODO (CM): Return a String, not a Result
+fn mk_revision_string() -> Result<String> { Ok(time::OffsetDateTime::now().format("%Y%m%d%H%M%S")) }
```
Comment (davidMcneil) [inline components/core/src/crypto/keys.rs]: Would it be possible to use `chrono`'s [`format`](https://docs.rs/chrono/0.4.10/chrono/struct.DateTime.html#method.format) here and then we could be completely rid of `time`?

File: components/sup-protocol/protocols/types.proto [hunk 2]
```diff
@@ -55,7 +55,7 @@ message PackageIdent {
 }
 
 message ProcessStatus {
-  optional int64 elapsed = 1;
+  optional uint64 elapsed = 1;
```
Comment (davidMcneil) [inline components/sup-protocol/protocols/types.proto]: Do we have to worry about breaking protocol compatability here?

File: components/core/src/crypto/keys.rs (hunk 1 above)
Comment (christophermaier) [reply components/core/src/crypto/keys.rs]: :thinking: Hrmm, that might be worth looking into. `chrono` actually wraps `time`, but maybe we could standardize on `chrono` to help simplify things; we do use `chrono` elsewhere, after all.

File: components/sup-protocol/protocols/types.proto (hunk 2 above)
Comment (christophermaier) [reply components/sup-protocol/protocols/types.proto]: I mentioned this in the commit message, but yeah, I think it should be fine:

https://developers.google.com/protocol-buffers/docs/proto#updating

File: components/core/src/crypto/keys.rs (hunk 1 above)
Comment (christophermaier) [reply components/core/src/crypto/keys.rs]: Actually, yes. We use `chrono` in just a small handful of places in the rest of the code, but it's always around formatting dates. I think it makes a lot of sense to standardize on `chrono` for that kind of stuff (Or standardize on `time`, for that matter!)

Comment (christophermaier) [reply components/core/src/crypto/keys.rs]: OK, we're on chrono, now!

**Pull request merged**